core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -92,12 +92,17 @@ sealed abstract class UnsupportedMediaTypeFailure(expected: Set[MediaRange]) ext
 final case class MediaTypeMissing(expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  val message = s"Decoder is unable to decode a Message without a MediaType. Expected media ranges: $expected"
+  val expectedMsg = expected.map(_.renderString).mkString(", ")
+  val message = s"No media type specified in Content-Type header. Expected one of the following media ranges: $expectedMsg"
 }
 
 /** Indicates that no [[EntityDecoder]] matches the [[MediaType]] of the [[Message]] being decoded */
 final case class MediaTypeMismatch(messageType: MediaType, expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  def message = s"$messageType is not a supported media type. Decoder accepts one of the following media ranges: $expected"
+  def message = {
+    val actual = messageType.renderString
+    val expectedMsg = expected.map(_.renderString).mkString(", ")
+    s"$actual is not a supported media type. Expected one of the following media ranges: $expectedMsg"
```

Comment:
So `alert("pnwed")` is an invalid `Content-Type` header, but I wonder if there is a way to craft a "valid" header that contains malicious javascript. What makes it valid? The fact that it was successfully parsed?

I am pretty cool with removing the input from the response since it should be pretty easy for a client to figure out what `Content-Type` header they are sending (you never know with proxies and what not), but it feels unfortunate that we have to provide a worse user experience just because some browser may decide to interpret `text/plain` as javascript (and execute it!).